l10n_br_stock_account/models/stock_picking.py

```diff
@@ -25,23 +25,20 @@ def _default_fiscal_operation(self):
 
             # Casos onde o usuário definiu o Pedido de Compra/Venda como
             # 'Sem Operação Fiscal'
-            if hasattr(self, "purchase_id") and hasattr(
-                self.purchase_id, "fiscal_operation_id"
-            ):
-                if not self.purchase_id.fiscal_operation_id:
+            try:
+                if self.purchase_id and not self.purchase_id.fiscal_operation_id:
                     fiscal_operation = False
-            if hasattr(self, "sale_id") and hasattr(
-                self.sale_id, "fiscal_operation_id"
-            ):
+                if self.sale_id and not self.sale_id.fiscal_operation_id:
+                    fiscal_operation = False
+            except AttributeError:
                 # Necessário para evitar o erro quando o l10n_br_purchase_stock
                 # é instalado:
                 # /l10n_br_stock_account/models/stock_picking.py", line 34,
                 # in _default_fiscal_operation
                 # if not self.sale_id.fiscal_operation_id:
                 # AttributeError: 'sale.order' object has no attribute
                 # 'fiscal_operation_id'
-                if not self.sale_id.fiscal_operation_id:
-                    fiscal_operation = False
+                pass
```

Comment:
@DiegoParadeda é preciso ver uma forma de corrigir sem usar o TRY, com os hasatrrs é melhor, porque isso passou a dar erro em versões recentes to PEP8:

* **[W8138(except-pass), PaymentOrder.generate_payment_file] pass into block except. If you really need to use the pass consider logging that exception**

lendo sobre o porque isso retorna erro pelo o que entendi é que sempre que acontecer "algo inesperado" o código precisar tratar esses casos e não apenas ignorar, uma referencia https://stackoverflow.com/questions/21553327/why-is-except-pass-a-bad-programming-practice , mas aqui não precisamos do TRY porque já sabemos os casos 

Estou vendo de corrigir isso no PR https://github.com/OCA/l10n-brazil/pull/3287 porém em outro modulo l10n_br_account_payment_brcobranca

Reply:
Na verdade dá pra fazer com o try/except, mas ai tem que por um logger explicativo ao invés do pass..
Eu particulamente prefiro a abordagem com o hasattr, o código fica mais simples.

Reply:
@antoniospneto pelo link e o debate feito lá é possível ver que a principal função do TRY é para ser usado quando programa não tem o Controle ou sabe a informação que será recebida, algo como uma consulta em um webservice ou um algum INPUT por equipamento ou usuário, assim como no outro PR não é esse o caso aqui nos já sabemos os possíveis erros e problemas e basta trata-los, não existe nenhum motivo aqui para incluir algo no LOG

Reply:
@mbcosta na verdade minha intenção no código não era substituir o has/get attr por try.
Era só remover o has e get, todos os testes que rodei funcionaram assim. Incluí o try realmente pensando que "algo inesperado" poderia poderia acontecer.

De qualquer forma vou tentar refazer com has/get, é que realmente já tentei e não funcionou

Reply:
Resumindo melhor a ideia, reparem que no código atual não existe nenhum erro aparente:

![image](https://github.com/user-attachments/assets/0389bbbe-a3df-413e-9d93-29b0a7461219)

Não faz sentido estar passando em nenhuma das linhas que setam fop = False (considerando uma transferência criada a partir do estoque) não é?

No entando não foi isso que observei no runbot, a fop sempre vem vazia.

Desconfio que seja algum problema do has/get attr em métodos "@api.model"

Reply:
Pessoal, deu certo, tive algum outro problema quando testei na primeira vez. Com hasattr funcionou sim

Reply:
@DiegoParadeda  eu vi esse erro de diferença entre um If

**if not purchase.fiscal_operation_id:**

que retorna False mesmo não existindo o self.purchase_id e precisa ser 

**if purchase and not purchase.fiscal_operation_id:**

Mas parece ter outros problemas quando deveria carregar o valor da Operação Fiscal do Pedido de Compra/Vendas na criação do stock.picking a partir desses objetos, parece que acontece porque o método default é chamado na criação de um novo picking porém mesmo tendo um self.purchase_id retorna False,  então como não encontrei uma forma de resolver isso parece que o melhor é ver outra forma de carregar a Operação Fiscal.

Estou vendo de inves de usar o método default colocar um onchange no campo invoice_state e a partir disso devolver a Operação Fiscal padrão, mas que caso tenha um Pedido de Compras/Vendas relacionado traga a Operação Fiscal informada no Pedido, parece que vai tornar possível tirar o hasattrs por poder herdar os métodos no l10n_br_sale_stock e l10n_br_purchase_stock, estou vendo de testar essas combinações para ver se é possível resolver com pouco código, se tiverem outras ideias e propostas de como resolver podemos ver de testar.